### PR TITLE
PLAT-502 Add segment refresh logic and helper tests

### PR DESCRIPTION
**Pull request by katekligman**
Adds the following:
* Expanded segmentation delete and refresh task logic
* Segment helper unit tests and bugfixes

### CODE REVIEW

File: segments/helpers.py [hunk 1]
```diff
@@ -96,8 +112,8 @@ def refresh_segment(self, segment_id, sql):
             self.remove_segment_membership(segment_id, user_id)
 
         # Copy the new adds and deletes to the member changed list
-        self.redis.sunionstore(new_key, self.segment_member_refresh_key, self.segment_member_refresh_key)
-        self.redis.sunionstore(del_key, self.segment_member_refresh_key, self.segment_member_refresh_key)
+        self.redis.sunionstore(self.segment_member_refresh_key, del_key)
```
Comment (Anton-Shutik) [inline segments/helpers.py]: Does it accept 2 params now ?

File: segments/models.py [hunk 2]
```diff
@@ -109,6 +115,12 @@ def do_refresh(sender, instance, created, **kwargs):
 signals.post_save.connect(do_refresh, sender=Segment)
 
 
+def do_delete(sender, instance, *args, **kwargs):
+    from segments.tasks import delete_segment
+    delete_segment.delay(instance.id)
+signals.post_delete.connect(do_delete, sender=Segment)
```
Comment (Anton-Shutik) [inline segments/models.py]:
```suggestion
@receiver(post_delete, sender=Segment)
def do_delete(sender, instance, *args, **kwargs):
    from segments.tasks import delete_segment
    delete_segment.delay(instance.id)
```

File: segments/tasks.py [hunk 3]
```diff
@@ -1,40 +1,33 @@
...
 import logging
+from celery import task
+from segments.helpers import SegmentHelper
+from segments.models import Segment, SegmentExecutionError
 
 logger = logging.getLogger(__name__)
 
 
-@task(name='segments_refresh')
+@task(name='refresh_segments')
```
Comment (Anton-Shutik) [inline segments/tasks.py]: Why do we need custom name here ?

File: segments/helpers.py (hunk 1 above)
Comment (katekligman) [reply segments/helpers.py]: I took it out entirely -- it's handled in the individual methods now.

**Pull request merged**